File: problem_in_multiple_languages/sol.java
// Programmer: The Viking
// Date: 19.08.2022


import java.util.*;

class sol {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter size of array: ");
        int n = sc.nextInt();
        System.out.print("Enter array: ");
        System.out.flush();
        int[] arr = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        HashMap<Integer,Integer> freq = new HashMap<Integer,Integer>();
        HashMap<Integer,Integer> ind = new HashMap<Integer,Integer>();
        for (int i = n - 1; i >= 0; i--) {
            ind.put(arr[i], i);
            int count = freq.containsKey(arr[i]) ? freq.get(arr[i]) : 0;
            freq.put(arr[i], count + 1);
        }
        List<ArrayList<Integer>> nums = new ArrayList<>();

        for (Map.Entry<Integer, Integer> entry : freq.entrySet()) {
            int v = entry.getKey();
            int f = entry.getValue();
            int i = ind.get(v);
            ArrayList<Integer> temp = new ArrayList<>();
            temp.add(-f);
            temp.add(i);
            temp.add(v);
            nums.add(temp);
        }
        Collections.sort(nums, new Comparator<ArrayList<Integer>>() {
            @Override
            public int compare(ArrayList<Integer> a, ArrayList<Integer> b) {
                if (a.get(0) != b.get(0)) {
                    return a.get(0) - b.get(0);
                } else {
                    return a.get(1) - b.get(1);
                }
            }
        });
        for (ArrayList<Integer> num : nums) {
            System.out.print(num.get(2) + " ");
        }
        System.out.print("\n");
    }
}

File: problem_in_multiple_languages/sol.js
// Programmer: The Viking
// Date: 19.08.2022


var readline = require('readline');

var rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

var arr = [];
rl.question("Enter array: ", function(inp) {
    arr = inp.split(' ').map(Number);
    rl.close();
});

rl.on('close', function() {
    var n = arr.length;
    var freq = {};
    var ind = {};
    for (var i = n - 1; i >= 0; i--) {
        ind[arr[i]] = i;
        if (freq[arr[i]]) {
            freq[arr[i]]++;
        } else {
            freq[arr[i]] = 1;
        }
    }
    var nums = [];
    for (const v in freq) {
        nums.push([-freq[v], ind[v], v]);
    }
    nums = nums.sort(function(a, b) {
        if (a[0] == b[0]) {
            return a[1] - b[1];
        } else {
            return a[0] - b[0];
        }
    });
    var ans = [];
    for (var i = 0; i < nums.length; i++) {
        ans.push(nums[i][2]);
    }
    console.log(...ans);
});
